core/src/main/java/uk/gov/gchq/magmacore/service/MagmaCoreService.java

```diff
@@ -724,9 +724,25 @@ public Thing getInTransaction(final IRI iri) {
      *
      * @param func {@link Function} to run.
      */
-    public void runInTransaction(final Function<MagmaCoreService, MagmaCoreService> func) {
+    public void runInReadTransaction(final Function<MagmaCoreService, MagmaCoreService> func) {
```

Comment:
Would it be worth making `TransactionType` a parameter of a single `runInTransaction` method to save code duplication?

Reply:
I did consider that but decided not to expose the `TxnType` class to the caller since it's Jena-specific and this class might need to be used for other RDF databases as well.  Some other classes do expose Jena classes but I hope we can change that in future if/when we add support for other databases. Would you agree?